shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -458,58 +462,51 @@ @interface FlutterClippingMaskViewPool ()
 // The maximum number of `FlutterClippingMaskView` the pool can contain.
 // This prevents the pool to grow infinately and limits the maximum memory a pool can use.
 @property(assign, nonatomic) NSUInteger capacity;
-@property(retain, nonatomic) NSMutableArray<FlutterClippingMaskView*>* pool;
-// The index points to the first available FlutterClippingMaskView in the `pool`.
-@property(assign, nonatomic) NSUInteger availableIndex;
+
+// The pool contains the views that are available to use.
+// The number of items in the pool must not excceds `capacity`.
+@property(retain, nonatomic) NSMutableSet<FlutterClippingMaskView*>* pool;
 
 @end
 
 @implementation FlutterClippingMaskViewPool : NSObject
 
 - (instancetype)initWithCapacity:(NSInteger)capacity {
   if (self = [super init]) {
-    _pool = [[NSMutableArray alloc] initWithCapacity:capacity];
+    _pool = [[NSMutableSet alloc] init];
```

Comment:
nit: `initWithCapacity` to save some hash extending operations?

Reply:
I have considered it and decided go against it.
For most of the pages, there are only 1 static PlatformViews. And even for scrolling situation, it is rear that many PlatformViews need to be clipped at the same time. So the set would rarely hit the capacity. 

Or maybe change it to `initWithCapacity:1`during initialization? I think in most cases, the PlatformView would have a clip.